libfreerdp/core/server.c

```diff
@@ -167,8 +167,20 @@ static BOOL wts_read_drdynvc_capabilities_response(rdpPeerChannel* channel, UINT
 	Stream_Seek_UINT8(channel->receiveData); /* Pad (1 byte) */
 	Stream_Read_UINT16(channel->receiveData, Version);
 	DEBUG_DVC("Version: %" PRIu16 "", Version);
-	channel->vcm->drdynvc_state = DRDYNVC_STATE_READY;
-	return TRUE;
+
+	if (Version < 1 || Version > 3)
```

Comment:
I think we should not filter out newer versions. In case MS adds version 4, it breaks our server implementations.

```suggestion
	if (Version < 1)
```

Reply:
Well in fact right now we only propose version 1 to the client, so we do not expect him to answer with something bigger.

Reply:
> Well in fact right now we only propose version 1 to the client, so we do not expect him to answer with something bigger.

In some MS protocols yes, but `[MS-RDPEDYC]` does exclude this situation here though:
```
The server indicates the maximum capability it supports, and the client responds with a Capabilities Response PDU that indicates the maximum capability that it can support.
```